src/main/java/org/asia/game/result/GameResultRepositoryFactory.java

```
@@ -0,0 +1,26 @@
+package org.asia.game.result;
+
+public class GameResultRepositoryFactory {
+
+    private final Destination destination;
+
+    public enum Destination {
+        IN_MEMORY,
+        FILE
+    }
+
+    public GameResultRepositoryFactory(Destination destination) {
+        this.destination = destination;
```

Please handle correctly the null parameter - we should throw an exception if someone passes `null` here instead of the correct enum.